interp/topconstr.ml

```diff
@@ -102,7 +102,8 @@ let fold_constr_expr_with_binders g f n acc = function
   | CAppExpl (loc,(_,_,_),l) -> List.fold_left (f n) acc l
   | CApp (loc,(_,t),l) -> List.fold_left (f n) (f n acc t) (List.map fst l)
   | CProdN (_,l,b) | CLambdaN (_,l,b) -> fold_constr_expr_binders g f n acc b l
-  | CLetIn (_,na,a,b) -> fold_constr_expr_binders g f n acc b [[na],default_binder_kind,a]
+  | CLetIn (_,na,a,t,b) ->
```

Comment:
I'm probably naive here, but I'm a bit surprised that `CLetIn` doesn't have the same structure (modulo the body) as `CProdN` and `CLambdaN`. After all, they can also carry a type or not, right? I.e. `fun x : T => ...` vs `fun x => ...`